keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/ClusterHandler.kt

```diff
@@ -117,43 +117,31 @@ class ClusterHandler(
       .getActiveServerGroups(resource)
       .byRegion()
 
-  override suspend fun willTakeAction(
-    resource: Resource<ClusterSpec>,
-    resourceDiff: ResourceDiff<Map<String, ServerGroup>>
-  ): ActionDecision {
-    // we can't take any action if there is more than one active server group
-    //  AND the current active server group is unhealthy
-    val potentialInactionableRegions = mutableListOf<String>()
-    val inactionableRegions = mutableListOf<String>()
-    resourceDiff.toIndividualDiffs().forEach { diff ->
-      if (diff.hasChanges() && diff.isEnabledOnly()) {
-        potentialInactionableRegions.add(diff.desired.location.region)
-      }
+  override fun getDesiredRegion(diff: ResourceDiff<ServerGroup>): String =
+    diff.desired.location.region
```

Comment:
can any of those be null or empty?

Reply:
no, other wise kotlin wouldn't let me do this